frontend/src/lib/api.ts

```diff
@@ -2119,10 +2089,36 @@ const api = {
     },
 }
 
-function reportError(method: string, url: string, response: Response, startTime: number): void {
-    const duration = new Date().getTime() - startTime
-    const pathname = new URL(url, location.origin).pathname
-    posthog.capture('client_request_failure', { pathname, method, duration, status: response.status })
+async function handleFetch(url: string, method: string, fetcher: () => Promise<Response>): Promise<Response> {
+    const startTime = new Date().getTime()
+
+    let response
+    let error
+    try {
+        response = await fetcher()
+    } catch (e) {
+        error = e
+    }
+
+    apiStatusLogic.findMounted()?.actions.onApiResponse(response, error)
+
+    if (error || !response) {
+        throw { status: 0, message: error ?? 'Unknown error' }
```

Comment:
really we should throw an actual error class here (and below)

Reply:
good idea